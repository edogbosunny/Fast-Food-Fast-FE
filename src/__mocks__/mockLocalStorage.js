let localStorage = {};

export default {
  setItem(key, value) {
    return Object.assign(localStorage, { [key]: value });
  },
  removeItem(key) {
    return delete localStorage[key];
  },
  getItem(key) {
    return localStorage[key];
  },
  clear() {
    localStorage = {};
  },
};
