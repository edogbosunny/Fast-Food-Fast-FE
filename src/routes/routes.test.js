/* eslint-disable no-underscore-dangle */
/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */
import React from 'react';
import { shallow, mount } from 'enzyme';
import 'jest-localstorage-mock';
import mockLocalStorage from '../__mocks__/mockLocalStorage';
import Routes from './routes';
// import setAuthToken from '../utils/setAuthToken';

describe('App', () => {
  window.localStorage = mockLocalStorage;
  const wrapper = shallow(
    <Routes />,
  );
  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should test local storage', () => {
    const KEY = 'jwtToken';
    const VALUE = 'nsssjjj';
    localStorage.setItem(KEY, VALUE);
    // console.log('?>?>?>?>', localStorage);

    // expect(localStorage.__STORE__[KEY]).toBe(VALUE);
  });
});
