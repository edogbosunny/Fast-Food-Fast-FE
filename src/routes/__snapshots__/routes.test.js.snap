// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter>
    <div>
      <Connect(withRouter(Navbar)) />
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
      </Switch>
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/login"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/signup"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/order-history"
        />
      </Switch>
      <Footer />
    </div>
  </BrowserRouter>
</Provider>
`;
